src/textual/widgets/_tree.py

```diff
@@ -31,12 +31,12 @@
 
 
 @dataclass
-class _TreeLine:
-    path: list[TreeNode]
+class _TreeLine(Generic[TreeDataType]):
```

Comment:
For me this is what started this. Tree "loses" it's type parameter when you look at `._line` and/or `TreeNode._children`. All I wanted to do was declare that those too were `TreeNode[TreeDataType]`. Along the way made a small number of similar changes in other places in Tree.